ota/common.js

```diff
@@ -107,10 +108,15 @@ async function requestOTA(endpoint) {
     }
 }
 
-function getImageBlockResponsePayload(image, imageBlockRequest) {
-    const start = imageBlockRequest.payload.fileOffset;
+function getImageBlockResponsePayload(image, imageBlockRequest, pageOffset, pageSize) {
+    const start = imageBlockRequest.payload.fileOffset + pageOffset;
     // When the data size is too big, OTA gets unstable.
-    let end = start + Math.min(50, imageBlockRequest.payload.maximumDataSize);
```

Comment:
I think we should only limit the datasize to 40 for insta devices (based on the manufacturerCode), lower data size = longer update time. Now this affects all devices.

Reply:
No problem - is the updated code in https://github.com/Koenkk/zigbee-herdsman-converters/pull/1082/commits/6f07dc2fa19eaad6c1b574e2958c52f76a3290f5 ok for you?